x-pack/plugins/ml/server/models/job_audit_messages/job_audit_messages.js

```diff
@@ -0,0 +1,235 @@
+            lte: 'now'
+          }
+        }
+      };
+    }
+
+    const query = {
+      bool: {
+        filter: [
+          {
+            bool: {
+              must_not: {
+                term: {
+                  level: 'activity'
+                }
+              }
+            }
+          },
+          timeFilter
+        ]
+      }
+    };
+
+    // if no jobId specified, load all of the messages
+    if (jobId !== undefined) {
+      query.bool.filter.push({
+        bool: {
+          should: [
+            {
+              term: {
+                job_id: '' // catch system messages
+              }
+            },
+            {
+              term: {
+                job_id: jobId // messages for specified jobId
+              }
+            }
+          ]
+        }
+      });
+    }
+
+    try {
+      const resp = await callWithRequest('search', {
+        index: ML_NOTIFICATION_INDEX_PATTERN,
+        ignore_unavailable: true,
+        size: 1000,
+        body:
+        {
+          sort: [
+            { timestamp: { order: 'asc' } },
+            { job_id: { order: 'asc' } }
+          ],
+          query
+        }
+      });
+
+      let messages = [];
+      if (resp.hits.total !== 0) {
+        messages = resp.hits.hits.map(hit => hit._source);
+      }
+      return messages;
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  // search highest, most recent audit messages for all jobs for the last 24hrs.
+  async function getAuditMessagesSummary() {
+    try {
+      const resp = await callWithRequest('search', {
+        index: ML_NOTIFICATION_INDEX_PATTERN,
+        ignore_unavailable: true,
+        size: 0,
+        body: {
+          query: {
+            bool: {
+              filter: {
+                range: {
+                  timestamp: {
+                    gte: 'now-1d'
+                  }
+                }
+              }
+            }
+          },
+          aggs: {
+            levelsPerJob: {
+              terms: {
+                field: 'job_id',
+              },
+              aggs: {
+                levels: {
+                  terms: {
+                    field: 'level',
+                  },
+                  aggs: {
+                    latestMessage: {
+                      terms: {
+                        field: 'message.raw',
+                        size: 1,
+                        order: {
+                          latestMessage: 'desc'
+                        }
+                      },
+                      aggs: {
+                        latestMessage: {
+                          max: {
+                            field: 'timestamp'
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      });
+
+      let messagesPerJob = [];
+      const LEVEL = { system_info: -1, info: 0, warning: 1, error: 2 };
```

Comment:
Can these be moved outside this function, so their values can be used in the `levelToText` function?